lib/spack/spack/detection/common.py

```diff
@@ -150,6 +155,29 @@ def executable_prefix(executable_dir):
     return os.sep.join(components[:idx])
 
 
+def library_prefix(library_dir):
+    """Given a directory where an library is found, guess the prefix
+    (i.e. the "root" directory of that installation) and return it.
+
+    Args:
+        library_dir: directory where an library is found
+    """
+    # Given a prefix where an library is found, assuming that prefix
+    # contains /lib/ or /lib64/, strip off the 'lib' or 'lib64' directory
+    # to get a Spack-compatible prefix
+    assert os.path.isdir(library_dir)
+
+    components = library_dir.split(os.sep)
+    if 'lib64' in components:
+        idx = components.index('lib64')
+        return os.sep.join(components[:idx])
+    elif 'lib' in components:
+        idx = components.index('lib')
+        return os.sep.join(components[:idx])
+    else:
+        return None
```

Comment:
If there is no `lib`/`lib64` to strip, then we should probably either:

* Return the library_dir as the prefix (this is what `executable_prefix` does
* Raise an exception (which is also reasonable but might lead to additional changes to decide when/if to address the exception) 

In the second case, that would probably involve replacing:

```
                pkg_prefix = library_prefix(prefix)

                 if not pkg_prefix:
                     msg = "no lib/ or lib64/ dir found in {0}. Cannot "
                     "add it as a Spack package"
                     llnl.util.tty.debug(msg.format(prefix))
                     continue
``` 

with a try/catch.